lib/src/realm_object.dart

```diff
@@ -141,6 +142,20 @@ class RealmCoreAccessor implements RealmAccessor {
   void set(RealmObject object, String name, Object? value, [bool isDefault = false]) {
     final propertyMeta = metadata[name];
     try {
+      if (value is List) {
+        if (value.isEmpty) {
+          return;
+        }
+
+        //This assumes the targe list property is empty. `valus is List` should happen only when making a RealmObject managed
```

Comment:
//This assumes the target list property is empty. `value is List` should happen only when making a RealmObject managed

Reply:
Could you elaborate?

Reply:
there is a typo. Fill fix

Reply:
fixed